ang/prospect/case-actions/services/unlink-contribution-case-action.service.js

```diff
@@ -0,0 +1,52 @@
+(function (angular) {
+  var module = angular.module('prospect');
+
+  module.service('UnlinkContributionCaseAction', UnlinkContributionCaseAction);
+
+  /**
+   * @param {object} ProspectConverted Prospect Converted Service
+   */
+  function UnlinkContributionCaseAction (ProspectConverted) {
+    /**
+     * Checks if the Action is allowed
+     *
+     * @param {object} action action
+     * @param {Array} cases cases
+     * @returns {boolean} if action is allowed
+     */
+    this.isActionAllowed = function (action, cases) {
+      if (!cases[0] || !cases[0].prospect.paymentInfo) {
+        return;
+      }
+
+      var isContributeTypeProspect =
```

Comment:
could we do an early return here? I mean:
```js
if (!isContributeTypeProspect) {
  return false;
}
```
and forget of the variable later. I try to avoid the long conditions like here, but it is just a suggestion :)